tests/react/async.test.tsx

```diff
@@ -394,20 +396,30 @@ it('updates an async atom in child useEffect on remount', async () => {
 
   await findByText('loading')
 
-  resolve.splice(0).forEach((fn) => fn())
-  await waitFor(() => {
+  await act(async () => {
     resolve.splice(0).forEach((fn) => fn())
-    getByText('count: 1')
   })
+  await findByText('count: 0')
 
-  fireEvent.click(getByText('button'))
+  await act(async () => {
```

Comment:
Oh!!!!!!! I now see what we were missing.

Reply:
For some reason, the same technique isn't working in `react@16.9.0` 🤔 

The async-derived atoms [work fine on CodeSandbox](https://codesandbox.io/s/jotai-async-test-16-9-0-oysspq) so it is confirmed to be an issue with the test.

Reply:
For `react@16.9.0`, does it work if we use node@14?

Reply:
We don't officially support `react@16.9.0` in Jotai v2.
So, whatever solution we are comfortable with should be good.

Reply:
No dice with Node V14 on local 😓. 

In that case, do we want to skip this test entirely for V16 and/or possibly drop 16 from the matrix?

Reply:
Yeah, seems like it's time to drop `16.9.0` from the matrix.

Reply:
Sooo it was a bug in `react@16.9.0`, bumping to react `react@16.14.0` fixes it 😁

Reply:
Cool!

Reply:
Do we still need node@16 and downgraded jsdom even with `react@16.14.0`?

Reply:
Unfortunately, we need both! 

Node 16 to resolve the [performance.mark errors](https://github.com/pmndrs/jotai/pull/1913#discussion_r1186527192). jsdom's latest version should work fine with v16 so [we could revert it](https://github.com/pmndrs/jotai/pull/1913/commits/b9e14f4c6e35d52953c3e3d85831c8e2f801036d).

Reply:
We need node 16 to resolve the [performance.mark errors](https://github.com/pmndrs/jotai/pull/1913#discussion_r1186527192). jsdom's latest version should work fine with v16 so [we could revert it](https://github.com/pmndrs/jotai/pull/1913/commits/b9e14f4c6e35d52953c3e3d85831c8e2f801036d).

Reply:
24c6291 Another hack.